test/acceptance/CheckoutSpec.scala

```diff
@@ -64,4 +109,218 @@ class CheckoutSpec extends FeatureSpec
+      val checkout = new pages.Checkout(testUser)
+      assert(checkout.pageHasLoaded())
+
+      And("I should be signed in with my Identity account")
+      assert(checkout.userDisplayName == testUser.username.toLowerCase)
+
+      And("first name, last name, and email address should be pre-filled")
+      assert(checkout.PersonalDetails.firstName.value == testUser.username)
+      assert(checkout.PersonalDetails.lastName.value == testUser.username)
+      assert(checkout.PersonalDetails.email.value == s"${testUser.username.toLowerCase}@gu.com")
+
+      And("I should have Identity cookies.")
+      go.to(checkout)
+      Seq("GU_U", "SC_GU_U", "SC_GU_LA").foreach { idCookie =>
+        assert(cookiesSet.map(_.getName).contains(idCookie))
+      }
+
+      When("I fill in the address")
+      checkout.fillInAddressDetails()
+
+      And("Select credit card payment option")
+      checkout.selectCreditCardPaymentOption()
+
+      And("I fill in credit card payment details")
+      checkout.fillInCreditCardPaymentDetails()
+
+      And("I submit the form")
+      checkout.submit()
+
+      Then("I should land on the Thank You page")
+      val thankYou = new ThankYou()
+      thankYou.pageHasLoaded()
+      assert(pageHasElement(name("subscriptionDetails")))
+
+      And("I should still be signed in with my Identity account.")
+      assert(thankYou.userDisplayName == testUser.username.toLowerCase)
+    }
+  }
+
+  feature("Checkout credit card failures") {
+    scenario("Charge is declined with a card_declined code", Acceptance) {
+      val checkout = new pages.Checkout(new TestUser)
+      When("I visit the checkout page ")
+      go.to(checkout)
+      assert(checkout.pageHasLoaded())
+
+      And("I fill in personal details")
+      checkout.fillInPersonalDetails()
+
+      And("Select credit card payment option")
+      checkout.selectCreditCardPaymentOption()
+
+      And("I fill in credit card payment details (card_declined)")
+      checkout.fillInCardDeclined()
+
+      And("I submit the form")
+      checkout.submit()
+
+      Then("the credit card should be declined")
+      assert(pageHasElement(id("payment-card-number")))
+
+      And("I remain on Checkout page")
+      assert(currentUrl(Config.driver) == checkout.url)
+    }
+
+    scenario("Charge will be declined with an incorrect_cvc code.", Acceptance) {
+      val checkout = new pages.Checkout(new TestUser)
+      When("I visit the checkout page ")
+      go.to(checkout)
+      assert(checkout.pageHasLoaded())
+
+      And("I fill in personal details")
+      checkout.fillInPersonalDetails()
+
+      And("Select credit card payment option")
+      checkout.selectCreditCardPaymentOption()
+
+      And("I fill in credit card payment details (incorrect_cvc)")
+      checkout.fillInCardDeclinedCvc()
+
+      And("I submit the form")
+      checkout.submit()
+
+      Then("the credit card should be declined")
+      assert(pageHasElement(id("payment-card-number")))
```

Comment:
and in fact, the only reliable way to distinguish unexpected 500 errors from expected Stripe errors might be to check the actual error message displayed in the payment details. Which would make the tests very brittle. Something worth considering for the next iteration, anyway